File: TaskManager/src/main/commands/Command.java
package main.commands;

import java.time.LocalDate;

import main.core.Issue;
import main.interfaces.IOperation;

enum Type {
    Create,
    Update,
    Delete
}

public abstract class Command implements IOperation{
    private String description;
    private LocalDate timeStamp;
    private Type type;

    public Command(String _description, Type _type) {
        this.description = _description,
        this.timeStamp = LocalDate.now();
        this.type = _type;
    }

    public abstract void execute();
    public abstract void undo();
}

File: TaskManager/TaskManager/Commands/Command.cs
using System;
namespace TaskManager.Commands;

public abstract class Command
{
    public abstract void Execute();
    public abstract void Undo();
    public string Description{ get; set; }
    public DateTime Timestamp { get; set; }
    public CommandType Type { get; set; }
}

public enum CommandType
{
    Create,
    Update,
    Delete
}